### ISSUE TITLE
WANTED: "Edit with Notepad++" translation

### ISSUE THREAD

**Issue by donho** [notepad-plus-plus/notepad-plus-plus#13201]
[New explorer context menu entry **"Edit with Notepad++"**](https://github.com/notepad-plus-plus/notepad-plus-plus/commit/a0031159cd15e2798996f53c05d98b10ec69ead0) has been added for Windows 11. This feature will be in the installer of next release. 

RP request: please find your language section to translate  **"Edit with Notepad++"** into your language in the following file:
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/installer/nativeLang/explorerContextMenuEntryLocal.ini

**Comment by rddim** [comment]
@donho 

Please, sort languages alphabetically

**Comment by ozone10** [comment]
Using `.gitattributes` file in root with
```
explorerContextMenuEntryLocal.ini   text diff working-tree-encoding=UTF-16LE-BOM eol=crlf
```

could help git process `explorerContextMenuEntryLocal.ini`  as text

**Comment by maisondasilva** [comment]
[brazilian_portuguese.xml]
Edit_with_Notepad++=Editar com Notepad++

**Comment by Regmos** [comment]
[danish.xml]
Edit_with_Notepad++=Redigér med Notepad++

**Comment by Edditoria** [comment]
[hongKongCantonese.xml]
Edit_with_Notepad++=用 Notepad++ 編輯

**Comment by donho** [comment]
> [hongKongCantonese.xml] Edit_with_Notepad++=用 Notepad++ 編輯

@Edditoria Unfortunately, it's limited by the language supported by NSIS installer.
- this entry is used when users chooses the language from the beginning of installation.
Thank you for your translation anyway.

**Comment by IoeCmcomc** [comment]
[vietnamese.xml]
Edit_with_Notepad++=Sửa bằng Notepad++

**Comment by GurliGebis** [comment]
@donho If the installer can write it some where in the registry, I'll update the handler to read it from there 🙂

**Comment by GurliGebis** [comment]
@donho that would be an easy way to do it 🙂

**Comment by SaiyajinK** [comment]
I've been waiting for this for a long time!
I use [Shell](https://github.com/moudey/Shell) to rename the entry and it works fine but natively it will be even better!

[**french.xml**]

`Edit_with_Notepad++=Éditer avec Notepad++`

or (with html special characters)

`Edit_with_Notepad++=&Eacute;diter avec Notepad++`

Can't wait for the translation to be integrated! Thanks ! 🙂

**Comment by GurliGebis** [comment]
I'm looking into supporting translation of it, so it is coming 🙂

**Comment by GurliGebis** [comment]
@donho I have a PR ready with english, danish and french - I just need to ensure language fallback is done, and we should be good to go.
If people have other translations, please just send them to me (or if there is a file with previous translations, I can add them as well).

**Comment by ArkadiuszMichalski** [comment]
Several translated entries have been added in [`explorerContextMenuEntryLocal.ini`](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/installer/nativeLang/explorerContextMenuEntryLocal.ini). Now this file is not needed anymore?

**Comment by GurliGebis** [comment]
@ArkadiuszMichalski correct - I have added them all to the resource file in the PR mentioned above.
The cool thing about having it as resources is that it changes if the user changes their UI language (or if they have two different users with different languages set up - once the multi user PR is merged).

**Comment by maisondasilva** [comment]
@GurliGebis I see it in English on mine is that expected right? My Brazilian Portuguese translation is above here https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13201#issuecomment-1444916932

![image](https://user-images.githubusercontent.com/11423362/230625367-e8f934b2-c4a6-460f-8d28-8825a74cab37.png)

**Comment by GurliGebis** [comment]
@maisondasilva yes, the translations hasn't been merged yet, so until then (and then the next release after that), it is hardcoded to english.

**Comment by maisondasilva** [comment]
@GurliGebis Obrigado pelo aviso! Fico muito bom agora!
I talked to the Winrar Developer he sent me this.

Hello,

I used this one: https://github.com/microsoft/AppModelSamples/tree/master/Samples/SparsePackages

as a sample.

First, you need to implement the shell extension using IExplorerCommand interface. IContextMenu is included only to "Show more options" menu in Windows 11.

Then you need to package this shell extension to .msix file, sign this file with digital signature and install this .msix package from another app.

Overall it is complicated and troublesome. I spent about two months to implement this and still searching for workarounds for some issues.

Eugene

**Comment by GurliGebis** [comment]
@maisondasilva thank you, it is good to hear they did the same.
The new implementation is done the same way.

**Comment by Mahadevan474** [comment]
Hi, 
I wanted to add for the Tamil translation. 
And wanted to let you know this is my first contribution. (Recently bought a new personal laptop :)
[tamil.xml]
Edit_with_Notepad++=Notepad++ பயன்படுத்தி தொகுக்க

**Comment by dracp** [comment]
Add Spanish translation in Explorer Context Menu
Notepad++ 8.5.2

Edit with Notepad = **Editar con Notepad++** 

Thnks.

![230508 NTP++8 5 2_Traducir_Edit-with_Notepad++_al español](https://user-images.githubusercontent.com/102802425/236909457-dc356acc-2976-4b10-9801-c40cf4293f5f.png)

**Comment by maisondasilva** [comment]
I created a new pull because although there was no "O" in my Brazilian translation, I got the same translation sent from Portugal to Portuguese. Acredito que no instalador está sendo enviado a variável Portuguese e não Portuguese Brazilian.

Can you check what happened and also accept the pull?

https://github.com/notepad-plus-plus/notepad-plus-plus/pull/13661

Thanks
Maison

**Comment by dracp** [comment]
In **Notepad++ 8.5.3**, "Edit with Notepad++" in context menu **is already shown in Spanish**. Thanks.

![230516 NTP++8 5 3_Edit-with-Notepad++_Traducido al español](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/102802425/efdaf8d5-fe77-45cb-926c-bd748dea5b2e)

**Comment by GurliGebis** [comment]
> Hi, I wanted to add for the Tamil translation. And wanted to let you know this is my first contribution. (Recently bought a new personal laptop :) [tamil.xml] Edit_with_Notepad++=Notepad++ பயன்படுத்தி தொகுக்க
> 
> Thanks

I have added your translation here: notepad-plus-plus/nppShell#31 , once that gets merged, it will be included in the next release after that.

**Comment by ner00** [comment]
@GurliGebis is it possible for the context menu to adopt the same language as the one set in the setup and preferences or is that a stretch at this point?

**Comment by gmikhail** [comment]
> Edit: please find your language section to translate **"Edit with Notepad++"** into your language in the following file:
> https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/installer/nativeLang/explorerContextMenuEntryLocal.ini and provide the translation here. I'll integrate it myself due to the repeated conflict.

Right now I get an error that the file is not found.

Anyway, here is my translations:
```
[russian.xml]
Edit_with_Notepad++=Изменить в Notepad++
```
```
[ukrainian.xml]
Edit_with_Notepad++=Редагувати в Notepad++
```
PS: I'm very happy that [this](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/7747) issue is finally resolved.

**Comment by GurliGebis** [comment]
> @GurliGebis is it possible for the context menu to adopt the same language as the one set in the setup and preferences or is that a stretch at this point?

It follows the language of the UI thread it is running under.
You can override it though, if needed.

**Comment by GurliGebis** [comment]
> > Edit: please find your language section to translate **"Edit with Notepad++"** into your language in the following file:
> > https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/installer/nativeLang/explorerContextMenuEntryLocal.ini and provide the translation here. I'll integrate it myself due to the repeated conflict.
> 
> Right now I get an error that the file is not found.
> 
> Anyway, here is my translations:
> 
> ```
> [russian.xml]
> Edit_with_Notepad++=Редактировать в Notepad++
> ```
> 
> ```
> [ukrainian.xml]
> Edit_with_Notepad++=Редагувати в Notepad++
> ```
> 
> PS: I'm very happy that [this](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/7747) issue is finally resolved.

That file has been removed, since it is not used for the translations anymore.
Just pasting the translations here is fine 🙂

**Comment by donho** [comment]
Anyway, here is my translations:

    [russian.xml]
    Edit_with_Notepad++=Редактировать в Notepad++

    [ukrainian.xml]
    Edit_with_Notepad++=Редагувати в Notepad++

    PS: I'm very happy that https://github.com/notepad-plus-plus/notepad-plus-plus/issues/7747 issue is finally resolved.

>That file has been removed, since it is not used for the translations anymore.
Just pasting the translations here is fine 🙂

@GurliGebis 
Please add them into the new PR:
https://github.com/notepad-plus-plus/nppShell/pull/31

**Comment by ner00** [comment]
> It follows the language of the UI thread it is running under.
> You can override it though, if needed.

That's exactly my point, it follows the language of the thread it is running under instead of the one set in the setup or general preferences. How do I override the context menu language? Maybe this is well documented, but I'm having trouble finding it in the user manual.

**Comment by GurliGebis** [comment]
@donho I will once I get home 😊
@ner00 you can override it by creating a text file inside the `contextMenu` folder, called `overrideMenuEntryLanguage.txt`
Then you just write the name of the language you want it to use in there, in the format like `en-US` and save the file.

**Comment by GurliGebis** [comment]
@donho I have updated the translation PR to include the ones above as well

**Comment by donho** [comment]
Thank you @GurliGebis!
 
There could be new translation in the future, so please keep updating this PR if there are new update entries coming.
I will merge the PR and do a new release of NppShell before doing a new release of Notepad++ (v8.5.4).

**Comment by GurliGebis** [comment]
@donho sounds like a plan 👍

**Comment by ghueller** [comment]
the text says
_Edit: please find your language section to translate "Edit with Notepad++" into your language in the following file:
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/installer/nativeLang/explorerContextMenuEntryLocal.ini and provide the translation here._
but the link is broken :/

**Comment by GurliGebis** [comment]
@donho can you update the description to just have people provide the translations in here? (I don't want people to edit .rc files manually, it is easier to just let me do it from VS and create a PR with it 🙂

**Comment by donho** [comment]
@GurliGebis 
Done

**Comment by schnurlos** [comment]
for german:
Mit Notepad++ bearbeiten

**Comment by victorel-petrovich** [comment]
**romanian.xml**
`Editează cu Notepad++`

**Comment by GurliGebis** [comment]
@victorel-petrovich added here: notepad-plus-plus/nppShell#42
@donho I have created a new translation branch, including the date in the name 🙂

**Comment by shard99** [comment]
**Norwegian**

These official languages/codes have the same translation:

Norwegian : nor (iso639-2) / no (iso-639-1)  = "Rediger i Notepad++"
Norwegian Nynorsk: nno (iso639-2) / nn (iso-639-1)  = "Rediger i Notepad++"
Norwegian Bokmål: nob (iso639-2) / nb (iso-639-1)  = "Rediger i Notepad++"

Someone else has to do the Sami languages...

**Comment by GurliGebis** [comment]
@shard99 I have created a pull request with your translations - once it is merged, it will be included in the next release thereafter.
